inventory_management_system_api/repositories/unit.py

```diff
@@ -86,3 +110,18 @@ def _is_duplicate_unit(self, code: str, session: ClientSession = None) -> bool:
         """
         logger.info("Checking if unit with code '%s' already exists", code)
         return self._units_collection.find_one({"code": code}, session=session) is not None
+
+    def _is_unit_in_catalogue_category(self, unit_id: str, session: ClientSession = None) -> bool:
+        """Checks if any documents in the database have a specific unit id
+
+        :param unit_id: The ID of the unit being looked for
+        :return: Returns True if 1 or more documents have the unit ID, false if none do
+        """
+        # Convert unit_id to ObjectId
+        unit_id = CustomObjectId(unit_id)
+
+        # Query for documents where 'unit_id' exists in the nested 'catalogue_item_properties' list
+        query = {"catalogue_item_properties.unit_id": unit_id}
+
+        # Check if any documents match the query
+        return self._catalogue_categories_collection.find_one(query, session=session) is not None
```

Comment:
Other comments are not needed, it's pretty self-explanatory.
```suggestion
        unit_id = CustomObjectId(unit_id)
        # Query for documents where 'unit_id' exists in the nested 'catalogue_item_properties' list
        query = {"catalogue_item_properties.unit_id": unit_id}
        return self._catalogue_categories_collection.find_one(query, session=session) is not None
```